test/common_test.go

```diff
@@ -319,6 +319,18 @@ func (c *T) RecvStart() bool {
 		return false
 	}
 	defer m.Free()
+	if addr := m.Port.Address(); addr != c.addr {
+		c.Debugf("Got unexpected message port address: %s", addr)
+		return false
+	}
+	if !m.Port.IsOpen() {
+		c.Debugf("Expected message port open")
+		return false
+	}
+	if m.Port.IsServer() || !m.Port.IsClient() {
```

Comment:
I suspect this is the part that is wrong.  Are you sure that you have client & server roles properly identified?  I'm not. :-)